plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -825,38 +815,57 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          };
       }
 
-      boost::asio::post(_thread_pool.get_executor(),
-                        [self = this, future{std::move(future)}, api_trx, is_transient, return_failure_traces,
-                         next{std::move(next)}, trx = trx]() mutable {
-                           if (future.valid()) {
-                              future.wait();
-                              app().executor().post(priority::low, exec_queue::read_write,
-                                 [self, future{std::move(future)}, api_trx, is_transient, next{std::move(next)}, trx{std::move(trx)},
-                                  return_failure_traces]() mutable {
-                                    auto start       = fc::time_point::now();
-                                    auto idle_time   = self->_time_tracker.add_idle_time(start);
-                                    auto trx_tracker = self->_time_tracker.start_trx(is_transient, start);
-                                    fc_tlog(_log, "Time since last trx: ${t}us", ("t", idle_time));
-
-                                    auto exception_handler =
-                                       [self, is_transient, &next, trx{std::move(trx)}, &start](fc::exception_ptr ex) {
-                                          self->log_trx_results(trx, nullptr, ex, 0, start, is_transient);
-                                          next(std::move(ex));
-                                       };
-                                    try {
-                                       auto result = future.get();
-                                       if (!self->process_incoming_transaction_async(result, api_trx, return_failure_traces, trx_tracker, next)) {
-                                          if (self->in_producing_mode()) {
-                                             self->schedule_maybe_produce_block(true);
-                                          } else {
-                                             self->restart_speculative_block();
-                                          }
-                                       }
-                                    }
-                                    CATCH_AND_CALL(exception_handler);
-                                 });
-                           }
-                        });
+      boost::asio::post( chain.get_thread_pool(),
```

Comment:
This section of code is deep nested and becomes complicated. Can more comments be added and/or it be refactored a bit?

Reply:
Cleaned it up some.